src/action/index.js

```diff
@@ -0,0 +1,291 @@
+      };
+   } else if (cartToken) {
+      return {
+         headers: {
+            "X-Cart-Token": localStorage.getItem("sc-cart-token"),
+         },
+      };
+   } else {
+      return null;
+   }
+}
+
+export const getAllProducts = () => async dispatch => {
+   try {
+      const resp = await axios.get(`${BASE_URL}/api/products`);
+      dispatch({
+         type: types.GET_ALL_PRODUCTS,
+         products: resp.data.products,
+      });
+   } catch (err) {
+      console.log(err);
+   }
+};
+
+export const getProductDetails = productId => async dispatch => {
+   try {
+      const response = await axios.get(`${BASE_URL}/api/products/${productId}`);
+      dispatch({
+         type: types.GET_PRODUCT_DETAILS,
+         product: response.data,
+      });
+   } catch (err) {
+      console.log(err);
+   }
+};
+
+export const clearProductDetails = () => ({ type: types.CLEAR_PRODUCT_DETAILS });
+
+export const addItemToCart = (productId, quantity) => async dispatch => {
+   try {
+      const axiosConfig = tokenCheck();
+
+      const resp = await axios.post(
+         `${BASE_URL}/api/cart/items/${productId}`,
+         {
+            quantity: quantity,
+         },
+         axiosConfig,
+      );
+
+      localStorage.setItem("sc-cart-token", resp.data.cartToken);
+
+      dispatch({
+         type: types.ADD_ITEM_TO_CART,
+         cartTotal: resp.data.total,
+      });
+   } catch (error) {
+      console.log("Add Item To Cart Error:", error.message);
+   }
+};
+
+export const getActiveCart = () => async dispatch => {
+   try {
+      const axiosConfig = tokenCheck();
+
+      const resp = await axios.get(`${BASE_URL}/api/cart`, axiosConfig);
+
+      dispatch({
+         type: types.GET_ACTIVE_CART,
+         cart: resp.data,
+      });
+   } catch (err) {
+      console.error(err);
+   }
+};
+
+export const getCartTotals = () => async dispatch => {
+   try {
+      const axiosConfig = tokenCheck();
+
+      const resp = axios.get(`${BASE_URL}/api/cart/totals`, axiosConfig);
+
+      dispatch({
+         type: types.GET_CART_TOTALS,
+         total: resp.data,
+      });
+   } catch (erro) {
+      console.error(erro);
+   }
+};
+
+export const createGuessOrder = guest => async dispatch => {
+   try {
+      const axiosConfig = tokenCheck();
+
+      const res = await axios.post(`${BASE_URL}/api/orders/guest`, guest, axiosConfig);
+
+      localStorage.removeItem("sc-cart-token");
+
+      dispatch({
+         type: types.CREATE_GUEST_ORDER,
+         order: {
+            id: res.data.id,
+            message: res.data.message,
+         },
+      });
+
+      return {
+         email: guest.email,
+         order_Id: res.data.id,
+      };
+   } catch (error) {
+      console.log("Error from Guess checkout", error);
+   }
+};
+
+export const getGuestOrderDetails = (email, orderId) => async dispatch => {
+   try {
+      const res = await axios.get(`http://api.sc.lfzprototypes.com/api/orders/guest/${orderId}?email=${email}`);
```

Comment:
Should have used the `BASE_URL` constant here instead of typing out the URL. If the API URL ever changes you want to have to only change it in one place.